### ISSUE TITLE
tracing: additional goroutine per rangefeed

### ISSUE THREAD

**Issue by erikgrinaker** [cockroachdb/cockroach#107053]
In 22.2, we see the number of goroutines per rangefeed increase from 4 to 5. The additional goroutine appears to be this:

```
1 @ 0x4907d6 0x4a071c 0x1653fb1 0x4c2401
# labels: {"dest_n":"1", "dest_r":"1012", "dest_s":"1", "distsql.gateway":"1", "f":"c975f0f6", "job":"883416806392332289", "n":"1"}
#	0x1653fb0	github.com/cockroachdb/cockroach/pkg/util/tracing/grpcinterceptor.newTracingClientStream.func2+0x70	github.com/cockroachdb/cockroach/pkg/util/tracing/grpcinterceptor/grpc_interceptor.go:340
```

Which comes from here:

https://github.com/cockroachdb/cockroach/blob/00fd3ef1fadde1d01f28a44d2bbeb65e3ba64f06/pkg/util/tracing/grpcinterceptor/grpc_interceptor.go#L348-L362

I believe this may have been introduced by #76306.

Jira issue: CRDB-29858

Epic CRDB-26372

**Comment by miretskiy** [comment]
Does this go routine matter much? Certainly won't with mux; not sure about regular rangefeed

**Comment by irfansharif** [comment]
Absolutely goroutine counts don't affect scheduling latencies, it's the runnable set. This goroutine is never runnable.

**Comment by erikgrinaker** [comment]
> Absolutely goroutine counts don't affect scheduling latencies, it's the runnable set. This goroutine is never runnable.

This is my intuition too, but they did see a significant latency degradation after upgrading to 22.2, presumably due to large-scale changefeeds (which is the only real workload on this cluster), so it seems worth doublechecking. Not expecting it to matter though.

**Comment by erikgrinaker** [comment]
Resolved by #107079.
